### PR TITLE
Support TLS 1.3 kexs and groups with DTLS 1.3

### PR DESCRIPTION
**Pull request by fwh-dc**

### CODE REVIEW

**Comment by fwh-dc** [comment]
@t8m This is ready for review. Would you please add labels to reflect that? Thanks!

**Comment by fwh-dc** [comment]
@t8m @mattcaswell Please review this before other DTLS 1.3 PR's as this is needed to pass tests in https://github.com/openssl/openssl/pull/22275. Thanks!

File: ssl/s3_lib.c [hunk 1]
```diff
@@ -3674,7 +3674,7 @@ long ssl3_ctrl(SSL *s, int cmd, long larg, void *parg)
         {
             unsigned int id;
 
-            if (SSL_CONNECTION_IS_TLS13(sc) && sc->s3.did_kex)
+            if ((SSL_CONNECTION_IS_TLS13(sc) || SSL_CONNECTION_IS_DTLS13(sc)) && sc->s3.did_kex)
```
Comment (mattcaswell) [inline ssl/s3_lib.c:3677]: We really need a `SSL_CONNECTION_IS_13` or similar....I thought you'd already added something like that in one of the previous PRs, but I can't now find it.

**Review by mattcaswell — CHANGES_REQUESTED** [review]

File: ssl/s3_lib.c (hunk 1 above)
Comment (fwh-dc) [reply ssl/s3_lib.c:3677]: I agree. It is part of this:
https://github.com/openssl/openssl/pull/22366

I'll rebase this PR and update once https://github.com/openssl/openssl/pull/22366 is merged.

**Comment by fwh-dc** [comment]
> #22366 is now merged. @fwh-dc - can you rebase this?

Done.

**Comment by mattcaswell** [comment]
Hmmm... what is that CI failure:

````
libcrypto.a(libcrypto-lib-defltprov.o): in function `deflt_get_params':
defltprov.c:(.text+0x12c): relocation truncated to fit: R_68K_GOT16O against `.LC0'
libcrypto.a(libcrypto-lib-pvkfmt.o): in function `derive_pvk_key.constprop.0':
pvkfmt.c:(.text+0x292): relocation truncated to fit: R_68K_GOT16O against `.LC[5](https://github.com/openssl/openssl/actions/runs/8552374677/job/23433349681?pr=22364#step:9:6)'
libcrypto.a(libcrypto-lib-x509_trust.o): in function `X509_TRUST_cleanup':
x509_trust.c:(.text+0x[6](https://github.com/openssl/openssl/actions/runs/8552374677/job/23433349681?pr=22364#step:9:7)34): relocation truncated to fit: R_68K_GOT16O against `trtable_free'
collect2: error: ld returned 1 exit status
````

Doesn't look relevant to this PR though.

**Review by mattcaswell — APPROVED** [review]

**Comment by t8m** [comment]
Not relevant, we would get rid of it if we rebased the feature branch 
against fresh master branch.

**Comment by t8m** [comment]
Squashed and merged to the feature branch. Thank you for your contribution.

**Pull request closed without merging**